### PR TITLE
Shared Feed: Filters

### PR DESCRIPTION
**Pull request by caiosba**
## Description

Implementing filters for the shared feed page when the feed and configured to share medias and requests. Nothing very new here, basically wiring existing components for filters and search fields.

Reference: CV2-4296.

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Manually.

## Things to pay attention to during code review

After:

Captura de tela de 2024-02-27 23-06-06

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Review by brianfleming — COMMENTED** [review]
@caiosba question on this, just to see if its a stale data/local dev thing, or something we can address:

- Looking at a few shared feeds I had created before, it look like the filters sometimes show the `Reset` or `Apply` buttons regardless of if anything has changed.
- Those buttons work different on the workspaces lists to only show up when appropriate

So is this just a data/local dev issue or something we can address?
Screenshot 2024-02-28 at 11 19 34 AM
Screenshot 2024-02-28 at 11 19 49 AM

**Comment by caiosba** [comment]
@brianfleming I didn't change the fact-checks view, but I'm going to review it for the medias/requests view.

**Comment by brianfleming** [comment]
> @brianfleming I didn't change the fact-checks view, but I'm going to review it for the medias/requests view.

Thanks for taking a look @caiosba 
Biggest motivation is to avoid any support costs, so questions don't come up with why these filter buttons don't behave the same as the lists, since they are presented as if they are identical

File: src/app/components/feed/FeedFilters.js [hunk 1]
```diff
@@ -163,7 +198,7 @@ const FeedFilters = ({
                 filterKey="range"
                 onChange={handleDateRange}
                 value={value || { request_created_at: {} }}
-                optionsToHide={['created_at', 'media_published_at', 'updated_at', 'report_published_at']}
+                optionsToHide={['created_at', 'media_published_at', 'report_published_at', 'request_created_at']}
```
Comment (dariusk) [inline src/app/components/feed/FeedFilters.js:201]: Oh this is good, `report_published_at` is so much more descriptive than `updated_at`

**Review by dariusk — APPROVED** [review]
Not too much to comment on this -- as you say, it's mostly gluing together existing components. I appreciate the declarative nature of the code changes.

**Comment by caiosba** [comment]
@brianfleming I fixed it and now both cases behave the same.

Also, thanks @dariusk for the review!

**Comment by brianfleming** [comment]
nice, thanks @caiosba 
Can you remind me how I can get my local version to see the new cards? I am on latest of all branches, and I created a new shared feed for both fact-checks and media, with a list of items, but I can't get it to render out locally. I think I remember you saying there is a script to run for now until the seed script is updated? Reason is, when talking to PMs tomorrow I want to be able to mess around locally to explain things.

**Comment by caiosba** [comment]
@brianfleming , let me ask Ashkan about how feasible it is to use the clusterization service locally... the other alternative is to use the seeds file, Manu is almost finishing it... let's follow-up on Slack.

**Pull request merged**